### PR TITLE
Fix documentation links

### PR DESCRIPTION
**Pull request by mfeurer**
This aims to fix broken links in the documentation. In order to do so it does:

* make warnings during documentation building fail the build process
* fix all warnings to make the build process pass
* automatically check for links to work with `make checklink`
* fix all links and update links for which a redirect was shown

In the process of creating this PR I opened #1046 ~~and #1047~~ so we can in the future fully automate the process of checking that ~~all links work~~ we have references to all objects in the API and examples.

### CODE REVIEW

**Review by PGijsbers — CHANGES_REQUESTED** [review]
Thanks! Automating checking for broken links is awesome :) 
Few questions:
 - I built the docs locally, but did not check all the links work - is that alright?
 - What stops us from referencing numpy docs to turn nitpicky on? 
 - Could you add this work to  `progress.rst`?

**Comment by mfeurer** [comment]
> I built the docs locally, but did not check all the links work - is that alright?

Yes, you need to explicitly do `make linkcheck`

> What stops us from referencing numpy docs to turn nitpicky on?

The fact that this already took me too long and I didn't have time any more ;) I think nitpicky also complained about the task type enum which I didn't know how to fix.

> Could you add this work to progress.rst?

Sure.

**Comment by PGijsbers** [comment]
The doc jobs fail.

**Comment by mfeurer** [comment]
> The doc jobs fail.

Yes, the rebase on #1044 made it fail :( I'll be working on that soonish, but probably not this week any more...

File: examples/20_basic/introduction_tutorial.py [hunk 1]
```diff
@@ -99,7 +99,7 @@
 # For this tutorial, our configuration publishes to the test server
 # as to not crowd the main server with runs created by examples.
 myrun = run.publish()
-print(f"kNN on {data.name}: http://test.openml.org/r/{myrun.run_id}")
+print(f"kNN on {data.name}: https://test.openml.org/r/{myrun.run_id}")
```
Comment (PGijsbers) [inline examples/20_basic/introduction_tutorial.py:102]:
```suggestion
print(f"kNN on {data.name}: {myrun.openml_url}")
```

File: examples/20_basic/simple_flows_and_runs_tutorial.py [hunk 2]
```diff
@@ -42,8 +42,8 @@
 # ==================
 
 myrun = run.publish()
-print("Run was uploaded to http://test.openml.org/r/" + str(myrun.run_id))
-print("The flow can be found at http://test.openml.org/f/" + str(myrun.flow_id))
+print("Run was uploaded to https://test.openml.org/r/" + str(myrun.run_id))
+print("The flow can be found at https://test.openml.org/f/" + str(myrun.flow_id))
```
Comment (PGijsbers) [inline examples/20_basic/simple_flows_and_runs_tutorial.py:46]:
```suggestion
print(f"Run was uploaded to {myrun.openml_url}")
print(f"The flow can be found at {myrun.flow.openml_url}")
```

File: examples/30_extended/flows_and_runs_tutorial.py [hunk 3]
```diff
@@ -69,7 +69,7 @@
 myrun = run.publish()
 # For this tutorial, our configuration publishes to the test server
 # as to not pollute the main server.
-print("Uploaded to http://test.openml.org/r/" + str(myrun.run_id))
+print("Uploaded to https://test.openml.org/r/" + str(myrun.run_id))
```
Comment (PGijsbers) [inline examples/30_extended/flows_and_runs_tutorial.py:72]:
```suggestion
print(f"Uploaded to {myrun.openml_url}")
```

File: examples/30_extended/flows_and_runs_tutorial.py [hunk 4]
```diff
@@ -115,7 +115,7 @@
 
 run = openml.runs.run_model_on_task(pipe, task, avoid_duplicate_runs=False)
 myrun = run.publish()
-print("Uploaded to http://test.openml.org/r/" + str(myrun.run_id))
+print("Uploaded to https://test.openml.org/r/" + str(myrun.run_id))
```
Comment (PGijsbers) [inline examples/30_extended/flows_and_runs_tutorial.py:118]:
```suggestion
print(f"Uploaded to {myrun.openml_url}")
```

File: examples/30_extended/flows_and_runs_tutorial.py [hunk 5]
```diff
@@ -159,7 +159,7 @@
 
 run = openml.runs.run_model_on_task(pipe, task, avoid_duplicate_runs=False, dataset_format="array")
 myrun = run.publish()
-print("Uploaded to http://test.openml.org/r/" + str(myrun.run_id))
+print("Uploaded to https://test.openml.org/r/" + str(myrun.run_id))
```
Comment (PGijsbers) [inline examples/30_extended/flows_and_runs_tutorial.py:162]:
```suggestion
print(f"Uploaded to {myrun.openml_url}")
```

File: examples/30_extended/flows_and_runs_tutorial.py [hunk 6]
```diff
@@ -229,7 +229,7 @@
 
     run = openml.runs.run_model_on_task(clf, task, avoid_duplicate_runs=False)
     myrun = run.publish()
-    print(f"kNN on {data.name}: http://test.openml.org/r/{myrun.run_id}")
+    print(f"kNN on {data.name}: https://test.openml.org/r/{myrun.run_id}")
```
Comment (PGijsbers) [inline examples/30_extended/flows_and_runs_tutorial.py:232]:
```suggestion
    print(f"kNN on {data.name}: {myrun.openml_url}")
```

**Review by PGijsbers — CHANGES_REQUESTED** [review]
Let's use the built-in function for generating openml urls. A bit less daunting and, at least as important, more future proof.

**Comment by mfeurer** [comment]
Great suggestion, just applied it!

**Review by PGijsbers — APPROVED** [review]
LGTM 🎉

**Pull request merged**